api/src/main/java/com/netflix/iceberg/io/FileIO.java

```diff
@@ -47,4 +46,30 @@
    * Delete the file at the given path.
    */
   void deleteFile(String path);
+
+  /**
+   * Get an {@link InputFile} to get the bytes for this table's metadata file with the given name.
+   */
+  InputFile readMetadataFile(String fileName);
+
+  /**
+   * Get an {@link OutputFile} to write bytes for a new table metadata file with the given name.
+   */
+  OutputFile newMetadataFile(String fileName);
```

Comment:
What do you think about renaming this to `newMetadataOutputFile` to be clear that this is producing an `OutputFile` and to fit the existing convention where `newInputFile` and `newOutputFile` are distinguished from one another?